src/matrix/compressed-matrix.cc

```diff
@@ -37,6 +37,17 @@ MatrixIndexT CompressedMatrix::DataSize(const GlobalHeader &header) {
   }
 }
 
+// scale all element of matrix by scaling floats
+// in GlobalHeader with alpha.
+void CompressedMatrix::Scale(float alpha) {
```

Comment:
I don't have objections per-se, but wasn't originally CompressedMatrix meant only to be immutable?